### PR TITLE
Fix possibly outdated onChange in color palette's color picker

### PR DESCRIPTION
**Pull request by costasovo**
## Description
ColorPalette component uses useCallback hook to memoize [ColorPicker render function](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/color-palette/index.js#L61-L70). This memoized renderer is not refreshed when onChange function changes. This causes that onChange function in ColorPicker may become outdated and carries outdated scope. Triggering such an outdated callback may cause issues because it is processed with potentially outdated scope. This PR adds ColorsPalette's onChange callback to the list of dependencies to the memoized  ColorPicker render so that the colour picker render method is refreshed after the onChange callback changes.

## How has this been tested?
I locally added a component to the ColorsPalette storybook to replicate the issue and checked that the fix works. Without the fix, you can observe that changing colour using the custom colour picker will reset previously set size value.

The component:
```
export const borderSettings = () => {
	const [ settings, setSettings ] = useState( {
		color: '#f00',
		size: 1,
	} );

	const colorsDef = [
		object( 'Red', { name: 'red', color: '#f00' } ),
		object( 'White', { name: 'white', color: '#fff' } ),
		object( 'Blue', { name: 'blue', color: '#00f' } ),
	];

	console.log(settings);

	return (
		<>
			<RangeControl
				value={ settings.size }
				onChange={ ( val ) => {
					setSettings( {
						...settings,
						size: val,
					} );
				} }
			/>
			<ColorPalette
				colors={ colorsDef }
				value={ settings.color }
				onChange={ ( val ) => {
					setSettings( {
						...settings,
						color: val,
					} );
				} }
			/>
		</>
	);
};
```

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/color-palette/index.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export default function ColorPalette( {
 				disableAlpha
 			/>
 		),
-		[ value ]
+		[ value, onChange ]
```
Comment (youknowriad) [inline packages/components/src/color-palette/index.js:69]: Can we just remove the useCallback call renderCustomColorPicker function. It seems to me that this optimization is not really necessary.

Comment (costasovo) [reply packages/components/src/color-palette/index.js:69]: Good point! 👍

Comment (costasovo) [reply packages/components/src/color-palette/index.js]: @youknowriad I've removed the useCallback and rebased the PR.

**Review by youknowriad — APPROVED** [review]
Looks good, thanks for the fix.

**Pull request merged**